account_statement_completion_compassion/model/completion_rules.py

```diff
@@ -0,0 +1,436 @@
+        if move_lines:
+            partner = move_lines[0].partner_id
+
+        if partner:
+            partner_obj = self.env['res.partner']
+            partner = partner_obj._find_accounting_partner(partner)
+            res['partner_id'] = partner.id
+
+        return res
+
+    def _generate_invoice(self, st_line, partner):
+        """ Genereates an invoice corresponding to the statement line read
+            in order to reconcile the corresponding move lines. """
+        # Read data in english
+        res = dict()
+        product = self.with_context(lang='en_US')._find_product_id(
+            st_line.ref)
+        if not product:
+            return res
+        # Don't gengerate invoice if it's a Sponsor gift
+        if product.categ_name == GIFT_CATEGORY:
+            res['name'] = product.name
+            contract_obj = self.env['recurring.contract']
+            contract_number = int(st_line.ref[16:21])
+            contract = contract_obj.search(
+                ['|',
+                 ('partner_id', '=', partner.id),
+                 ('correspondant_id', '=', partner.id),
+                 ('num_pol_ga', '=', contract_number),
+                 ('state', '!=', 'draft')])
+            if len(contract) == 1:
+                # Retrieve the birthday of child
+                birthdate = ""
+                if product.name == GIFT_NAMES[0]:
+                    birthdate = contract.child_id.birthdate
+                    res['child_birthdate'] = birthdate
+                    birthdate = datetime.strptime(birthdate, DF).strftime(
+                        "%d %b")
+                res['name'] += "[" + contract.child_id.code
+                res['name'] += " (" + birthdate + ")]" if birthdate else "]"
+            else:
+                res['name'] += " [Child not found] "
+            return res
+
+        # Setup invoice data
+        invoicer_id = st_line.statement_id.recurring_invoicer_id.id
+        if not invoicer_id:
+            invoicer_id = self.env['recurring.invoicer'].create(
+                {'source': st_line.statement_id._name}).id
+            st_line.statement_id.write({'recurring_invoicer_id': invoicer_id})
+        inv_data = {
+            'account_id': partner.property_account_receivable.id,
+            'type': 'out_invoice',
+            'partner_id': partner.id,
+            'journal_id': st_line.statement_id.journal_id.id,
+            'date_invoice': st_line.date,
+            'payment_term': 1,  # Immediate payment
+            'bvr_reference': st_line.ref,
+            'recurring_invoicer_id': invoicer_id,
+        }
+
+        # Create invoice and generate invoice lines
+        invoice_obj = self.env['account.invoice']
+        invoice = invoice_obj.with_context(lang='en_US').create(inv_data)
+
+        res.update(self._generate_invoice_line(
+            invoice.id, product, st_line, partner.id))
+
+        if product.categ_name != GIFT_CATEGORY:
```

Comment:
No, removing completely the test and doing following lines in all cases. At line 283, you return if categ_name == GIFT_CATEGORY so here categ_name will never be equal to GIFT_CATEGORY.